examples/cpp/PineAPPL.hpp

```diff
@@ -0,0 +1,257 @@
+/**
+ * @file PineAPPL.hpp
+ * @brief object oriented interface to PineAPPL
+ */
+#ifndef PineAPPL_HPP_
+#define PineAPPL_HPP_
+
+#include <string>
+#include <list>
+#include <LHAPDF/LHAPDF.h>
+
+#include <pineappl_capi.h>
+
+/** @brief object oriented interface to PineAPPL  */
+namespace PineAPPL {
+
+/** @brief Key-value storage for passing optional information during grid creation */
+struct KeyVal {
+    /** @brief underlying raw object */ 
+    pineappl_keyval *raw;
```

Comment:
I'd like to keep it public, because
- this way one can always fallback to the C interface (e.g. if I'm missing a function)
- and otherwise I have to make all classes friends with each other (which is unnecessarily complicated)
- and I can keep everything public (see https://github.com/N3PDF/pineappl/pull/178/files#r974458714)